src/datascience-ui/data-explorer/mainPanel.tsx

```diff
@@ -218,37 +232,42 @@ export class MainPanel extends React.Component<IMainPanelProps, IMainPanelState>
                     gridRows: initialRows,
                     totalRowCount,
                     fetchedRowCount: initialRows.length,
-                    indexColumn: indexColumn
+                    indexColumn: indexColumn,
+                    originalVariableShape: variable.originalVariableShape ?? variable.shape,
+                    dataDimensionality: variable.dataDimensionality ?? 2,
+                    isSliceDataSupported: variable.isSliceDataSupported,
+                    // Maximum number of rows is 100 if evaluating in debugger, undefined otherwise
+                    maximumRowChunkSize: variable.maximumRowChunkSize ?? this.state.maximumRowChunkSize
                 });
 
+                // New data coming in, so reset everything and clear our cache of columns
+                this.columnsContainingInfOrNaN.clear();
+                this.resetGridEvent.notify();
```

Comment:
There's something very weird happening here where the filter box does not render on a grid reset unless it was already set to visible before. Still working on tracking this down.